src/Mod/TechDraw/App/Cosmetic.h

```diff
@@ -95,25 +115,19 @@ class TechDrawExport CosmeticEdge : public Base::Persistence, public TechDraw::B
     double permaRadius;
 //    void unscaleEnds(double scale);
     TechDraw::BaseGeomPtr m_geometry;
-    LineFormat m_format;
-
-    boost::uuids::uuid getTag() const;
-    std::string getTagAsString() const override;
 
 protected:
     //Uniqueness
-    void createNewTag();
     void assignTag(const TechDraw::CosmeticEdge* ce);
-    boost::uuids::uuid tag;
-
-    Py::Object PythonObject;
+    using Cosmetic::createNewTag;
 
+    using Cosmetic::tag;
 };
 
 //********** GeomFormat ********************************************************
 
 // format specifier for geometric edges (Edge5)
-class TechDrawExport GeomFormat: public Base::Persistence
+class TechDrawExport GeomFormat: public TechDraw::Cosmetic
```

Comment:
Why would GeomFormat have Cosmetic as an ancestor?  GeomFormat is not a specialization of Cosmetic.   It is a LineFormat + a geometry index.

Reply:
`Cosmetic` shouldn't be understood in terms of whether it's part of OCC or artificially made by the user. `Cosmetic` should be understood as an object that holds visual data. And we want the objects using `LineFormat` to use dynamic dispatch all the locations were we need to access `LineFormat` without having lots of `bg->source()` trees.

So essentially, GeomFormat inherits TechDraw::Cosmetic because it has been treated throughout the TechDraw code in the same way as CenterLine, CosmeticEdge. It has to do with GeomFormats usage inside the TechDraw code rather than what we as people (without knowledge of TechDraw internals) define as a cosmetic object

Here we see how GeomFormat in the exact same way as CenterLine and CosmeticEdge internally by the code. Therefore we want to see them as the same type and use dynamic dispatch for all of them. Regarding the code, `Cosmetic` should be understood as an object that holds visual data, not as something artificially made by the user.

https://github.com/FreeCAD/FreeCAD/blob/7ba145c36f161d0b25a7f6848cde620e06286cf7/src/Mod/TechDraw/Gui/TaskLineDecor.cpp#L111-L132